bser/decode_test.go

```diff
@@ -770,6 +779,24 @@ var decodeBenches = map[string]decodeBench{
 			return dst, err
 		},
 	},
+	"raw_message_slice": {
+		encoded: []byte(
+			"\x00\x01\x05\x1c\x00\x00\x00\x01\x03\x01\x02\x03\x06Values\x00\x03\x02\x02\x03\x02ok\x02\x03\x05there",
+		),
+		doDecode: func(dec *Decoder) (interface{}, error) {
+			var v RawMessage
+			err := dec.Decode(&v)
+			return v, err
+		},
+	},
+	"custom_unmarshaler": {
+		encoded: []byte("\x00\x01\x03\x02\x03\x7b"),
+		doDecode: func(decoder *Decoder) (interface{}, error) {
```

Comment:
do these need a expected data?

Reply:
@jonasi that’s a good question. I generally tend to leave checking for expected results to the tests (which seems to be what the [standard library does in a similar  case](https://golang.org/src/encoding/json/bench_test.go#L248)).

The pros of this approach is that it reduces the amount of code necessary to add a benchmark (don’t need to add expected values or start/stop the timer before checking) and that it helps separate the responsibilities of tests and benchmarks.

The cons are that your benchmarks might not be measuring what you think they are, but then again you should have tests to check this.

I don’t feel particularly strongly either way, so if you think these benchmarks should also check the results it won’t be difficult to change.

Reply:
ah - missed that these were benchmark items.   you're right.